src/overlays/actors/ovl_En_Tanron4/z_en_tanron4.c

```diff
@@ -28,22 +44,241 @@ const ActorInit En_Tanron4_InitVars = {
+    thisx->speedXZ = 3.0f + KREG(48);
+    thisx->uncullZoneForward = 10000.0f + KREG(70);
+    this->randRollTimer = Rand_ZeroFloat(10.0f);
+
+    if (thisx->params == SEAGULL_FOLLOW_ACTOR) {
+        EnTanron4_SetupFlyNearActor(this);
+    } else {
+        EnTanron4_SetupFlyNearHome(this);
+
+        // spawn number of seagulls specified by params
+        if (thisx->params >= 0) {
+            s32 i;
+
+            for (i = 0; i < thisx->params; i++) {
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_TANRON4,
+                            thisx->world.pos.x + randPlusMinusPoint5Scaled(500.0f),
+                            thisx->world.pos.y + randPlusMinusPoint5Scaled(100.0f),
+                            thisx->world.pos.z + randPlusMinusPoint5Scaled(500.0f), 0, Rand_ZeroFloat(65536.0f), 0,
+                            SEAGULL_CLONE);
+            }
+        }
+
+        if ((gSaveContext.time > CLOCK_TIME(20, 0)) || (gSaveContext.time < CLOCK_TIME(4, 0))) {
+            this->timeInfluence = 1500.0f;
+            thisx->world.pos.y += 1500.0f;
+        }
+    }
+}
+
+void EnTanron4_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnTanron4_SetupFlyNearHome(EnTanron4* this) {
+    this->actionFunc = EnTanron4_FlyNearHome;
+    Animation_MorphToLoop(&this->skelAnime, &gSeagullFlapAnim, 0.0f);
+}
+
+void EnTanron4_FlyNearHome(EnTanron4* this, GlobalContext* globalCtx) {
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    f32 distToTarget;
+    s16 xRot;
+    s16 yRot;
+    s16 zRot;
+
+    // `timeInfluence` controls both the height of the seagulls and when they are visible.
+    // They fly higher in the sky as the night goes on, and they dissapear as dawn approaches.
+    if ((gSaveContext.time > CLOCK_TIME(20, 0)) || (gSaveContext.time < CLOCK_TIME(4, 0))) {
+        Math_ApproachF(&this->timeInfluence, 1500.0f, 1.0f, 1.0f);
+    } else {
+        Math_ApproachZeroF(&this->timeInfluence, 1.0f, 1.0f);
+    }
+
+    xDiff = this->targetPos.x - this->actor.world.pos.x;
+    yDiff = (this->targetPos.y + this->timeInfluence) - this->actor.world.pos.y;
+    zDiff = this->targetPos.z - this->actor.world.pos.z;
+
+    distToTarget = sqrtf(SQ(xDiff) + SQ(zDiff));
+
+    if ((this->timers[SEAGULL_TIMER_CHOOSE_TARGET] == 0) || (distToTarget < 100.0f)) {
+        this->targetPos.x = this->actor.home.pos.x + randPlusMinusPoint5Scaled(500.0f);
+        this->targetPos.y = this->actor.home.pos.y + randPlusMinusPoint5Scaled(100.0f);
+        this->targetPos.z = this->actor.home.pos.z + randPlusMinusPoint5Scaled(500.0f);
+
+        this->timers[SEAGULL_TIMER_CHOOSE_TARGET] = Rand_ZeroFloat(100.0f) + 60.0f;
+        this->step = 0;
+    }
+
+    yRot = Math_Atan2S(xDiff, zDiff);
+    xRot = Math_Atan2S(yDiff, distToTarget);
+    zRot = Math_SmoothStepToS(&this->actor.world.rot.y, yRot, 0xA, this->step, 0);
+
+    if (zRot > 0x1000) {
+        zRot = 0x1000;
+    } else if (zRot < -0x1000) {
+        zRot = -0x1000;
+    }
```

Comment:
I assume this doesn't match with a CLAMP macro.
I'm not sure if it would be better tho, up to you

Reply:
yea it didnt match. the branching is backwards